### DISCUSSION TITLE
How to fill in fields using useFieldArray ?

### DISCUSSION THREAD

**Discussion by DanPresa** [react-hook-form/react-hook-form#10006] [Q&A]
I want to fill out a dynamic form but. I used useFieldArray and now. I don't know how to fill in each field with data that comes from an API.
Is it possible to do that?

**Comment by Moshyfawn** [comment]
With async data, if you call `useForm` before you have the data, you probablu need to
- **either**
```js
const data = useAsyncData()
const { reset } = useForm()

useEffect(() => {
  if (data) reset(data)
}, [reset, data])
```
- **or** use [the new `values` `useForm` API](https://react-hook-form.com/api/useform#props)
